x-pack/plugin/fleet/src/main/java/org/elasticsearch/xpack/fleet/action/PostSecretRequest.java

```diff
@@ -50,6 +50,12 @@ public void writeTo(StreamOutput out) throws IOException {
 
     @Override
     public ActionRequestValidationException validate() {
+        if (this.source == null || !this.source.contains("\"value\":")) {
```

Comment:
@ioanatia added validation here, it might be better to actually parse the source as json first, WDYT?

Reply:
yes, we want to parse the JSON.



An example:

We need a PARSER in the `PostSecretRequest` and a static method `parseRequest` method, like here:

https://github.com/elastic/elasticsearch/blob/295bdbf9e6e1f5232f32821c2d5c3c76a834ec4c/modules/data-streams/src/main/java/org/elasticsearch/datastreams/lifecycle/action/PutDataStreamLifecycleAction.java#L53-L64

then in the rest handler, this is how the request object is created:

https://github.com/elastic/elasticsearch/blob/295bdbf9e6e1f5232f32821c2d5c3c76a834ec4c/modules/data-streams/src/main/java/org/elasticsearch/datastreams/lifecycle/rest/RestPutDataStreamLifecycleAction.java#L42

in our case, we would need to call that static parse method here:

https://github.com/elastic/elasticsearch/blob/827e9c4fe4afed945925235236b906390dbb6dfa/x-pack/plugin/fleet/src/main/java/org/elasticsearch/xpack/fleet/rest/RestPostSecretsAction.java#L40

Reply:
@ioanatia I tried this but I can't figure it out. See this commit: https://github.com/elastic/elasticsearch/pull/97728/commits/acf4835725488b6d1a9cba0b79cb480069070cb8
When I add the parser, and set `PostSecretRequest.source` as the secret value, it fails at [TransportPostSecretAction.doExecute](https://github.com/elastic/elasticsearch/blob/f299539adbbe5d2a2164f6dd53d01ffce64e9e14/x-pack/plugin/fleet/src/main/java/org/elasticsearch/xpack/fleet/action/TransportPostSecretAction.java#L34) because source is expected to be a JSON. So I would need to transform the value back to JSON, which I tried with `XContentBuilder` but I didn't find how to get the string representation from it.

```
    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
        builder.startObject();
        builder.field(VALUE_FIELD.getPreferredName(), this.value);
        builder.endObject();
        return builder;
    }
```

Reply:
This is a common pattern that's being used to parse the content and to have required parameters.
We need validation here to require `value` and to make sure `value` is a string.
It's okay with me if we want to follow up on this in another PR. Same with the privilege yml tests.

You are right with this change `PostSecretRequest.source` is actually the `value` field, but then we use it here as a JSON:

https://github.com/elastic/elasticsearch/blob/731cd6ff915b20ec9f78769c3d7e0a5204209aa8/x-pack/plugin/fleet/src/main/java/org/elasticsearch/xpack/fleet/action/TransportPostSecretAction.java#L34

so this would need a bit more refactoring in order to work, and it's perfectly fine to me to do it separately.

Reply:
okay, reverted to the last passing commit and will add the changes to a new pr